crates/extensions/tedge_log_manager/src/tests.rs

```diff
@@ -0,0 +1,582 @@
+        .with_raw_content(&format!(
+            r#"files = [
+            {{ type = "type_one", path = "{tempdir_path}/file_a" }},
+            {{ type = "type_one", path = "{tempdir_path}/file_b" }},
+            {{ type = "type_two", path = "{tempdir_path}/file_c" }},
+            {{ type = "type_one", path = "{tempdir_path}/file_d" }},
+        ]"#
+        ));
+
+    Ok(tempdir)
+}
+
+/// Create a log manager actor builder
+/// along two boxes to exchange MQTT and HTTP messages with the log actor
+#[allow(clippy::type_complexity)]
+fn new_log_manager_builder(
+    temp_dir: &Path,
+) -> (
+    LogManagerBuilder,
+    TimedMessageBox<SimpleMessageBox<MqttMessage, MqttMessage>>,
+    SimpleMessageBox<HttpRequest, HttpResult>,
+    SimpleMessageBox<NoMessage, FsWatchEvent>,
+) {
+    let config = LogManagerConfig {
+        config_dir: temp_dir.to_path_buf(),
+        topic_root: "te".to_string(),
+        topic_identifier: "device/main//".to_string(),
+        plugin_config_dir: temp_dir.to_path_buf(),
+        plugin_config_path: temp_dir.join("c8y-log-plugin.toml"),
+        logtype_reload_topic: Topic::new_unchecked("te/device/main///cmd/log_upload"),
+        logfile_request_topic: TopicFilter::new_unchecked("te/device/main///cmd/log_upload/+"),
+        current_operations: HashSet::new(),
+    };
+
+    let mut mqtt_builder: SimpleMessageBoxBuilder<MqttMessage, MqttMessage> =
+        SimpleMessageBoxBuilder::new("MQTT", 5);
+    let mut http_builder: SimpleMessageBoxBuilder<HttpRequest, HttpResult> =
+        SimpleMessageBoxBuilder::new("HTTP", 1);
+    let mut fs_watcher_builder: SimpleMessageBoxBuilder<NoMessage, FsWatchEvent> =
+        SimpleMessageBoxBuilder::new("FS", 5);
+
+    let log_builder = LogManagerBuilder::try_new(
+        config,
+        &mut mqtt_builder,
+        &mut http_builder,
+        &mut fs_watcher_builder,
+    )
+    .unwrap();
+
+    (
+        log_builder,
+        mqtt_builder.build().with_timeout(TEST_TIMEOUT_MS),
+        http_builder.build(),
+        fs_watcher_builder.build(),
+    )
+}
+
+/// Create a log manager actor ready for testing
+fn new_log_manager_actor(temp_dir: &Path) -> LogManagerActor {
+    let (actor_builder, _, _, _) = new_log_manager_builder(temp_dir);
+    actor_builder.build()
+}
+
+/// Spawn a log manager actor and return 2 boxes to exchange MQTT and HTTP messages with it
+fn spawn_log_manager_actor(
+    temp_dir: &Path,
+) -> (
+    MqttMessageBox,
+    SimpleMessageBox<HttpRequest, HttpResult>,
+    SimpleMessageBox<NoMessage, FsWatchEvent>,
+) {
+    let (actor_builder, mqtt, http, fs) = new_log_manager_builder(temp_dir);
+    let mut actor = actor_builder.build();
+    tokio::spawn(async move { actor.run().await });
+    (mqtt, http, fs)
+}
+
+#[test]
+/// Filter on type = "type_one".
+/// There are four logs created in tempdir { file_a, file_b, file_c, file_d }
+/// Of which, { file_a, file_b, file_d } are "type_one"
+fn test_filter_logs_on_type() {
```

Comment:
We don't need this test as long as `filter_logs_on_type` is tested in `log_utils.rs`.